frontend/src/services/ApiService.ts

```diff
@@ -78,3 +78,34 @@ ApiService.interceptors.response.use(
     throw transformedError
   },
 )
+
+export const processFetchResponse = async (response: Response) => {
+  try {
+    const data = await response.json()
+
+    // TODO: data may be present but if response is non 2XX, transform to error and throw
+    return data
+  } catch (error) {
+    if (error instanceof Error) {
```

Comment:
This doesn't look correct? Why do we need to check the type of `error`?

Reply:
[541bf25](https://github.com/opengovsg/FormSG/pull/6008/commits/541bf25cefe0bd9526f7197b32409e7958124cf8)